### PR TITLE
[relay][pass manager] Open transform namespace

### PR DESCRIPTION
**Pull request by zhiics**
This PR is the first item of #3202 

Now we open the `transform` namespace by adding transform.h and transform.py. Currently, pass manager related code is moved to these two files. The ir_pass related code is left in the original place. They will be moved once we finish porting.

cc @jroesch @tqchen @MarisaKirisame @wweic @slyubomirsky

### CODE REVIEW

**Comment by zhiics** [comment]
I am also going to change `sequential_pass` and `SequentialPass` to `sequential`. That is what we intended, right? @tqchen @MarisaKirisame

**Comment by tqchen** [comment]
We don’t want Sequential and sequential, both of which are duplicated. 

Consider merge as Sequential(use constructor).

**Comment by tqchen** [comment]
A quick way to check is to see if we can reproduce the example shown in the RFC

**Comment by zhiics** [comment]
I think we should be able to execute that example using the current Sequential. The problem is that we need to finish pass registration, meaning we need to have `Pass FoldConstant` first. This is the goal of #3191, @MarisaKirisame is working on it.

File: include/tvm/relay/transform.h [hunk 1]
```diff
@@ -182,6 +183,122 @@ class Pass : public NodeRef {
...
+
+/*!
+ * \brief The SequentialNode contains a set of passes that transform Relay
+ * programs from one AST to another semantically equivalent one.
+ *
+ * One example of this level of pass is that the pass manager needs to correctly
+ * perform a host of optimizations with a given optimization level and disabled
+ * passes.
+ */
+class SequentialNode : public PassNode {
```
Comment (tqchen) [inline include/tvm/relay/transform.h]: declare node, but hide the implementation inside. (User don't need to see SequentialNode, they only need to see Sequantial)

**Comment by tqchen** [comment]
Merge this in for now. Thanks, @zhiics. I will propose a few improves in a follow-up PR

**Pull request merged**